### PR TITLE
Replaced sdl1.2 with OpenTK for Joystick and Gamepad

### PR DESCRIPTION
**Pull request by cra0zy**
#3407 looked too ugly... and I still have to learn how to rebase. Also this PR provides Gamepad class from #3119, but a little bit modified.

### CODE REVIEW

**Comment by tomspilman** [comment]
What platforms has this been tested on?

Also after we merge this, can you follow up with a second PR organizing:

  - `Axis`
  - `Capabilities`
  - `Input`
  - `PadConfig`
  - `Joystick`
  - `Stick`

These types seem to be internals to the OpenTK `GamePad` implementation.  I don't think we want them public.  We also probably want to move them into some other namespace like `Microsoft.Xna.Framework.Input.OpenTK` or something.

File: Build/Projects/FrameworkReferences.definition [hunk 1]
```diff
@@ -26,6 +26,9 @@
       <Binary
         Name="Tao.Sdl"
         Path="ThirdParty/GamepadConfig/Tao.Sdl.dll" />
```
Comment (tomspilman) [inline Build/Projects/FrameworkReferences.definition]: What is "Tao.Sdl.dll" used for here now?

Comment (cra0zy) [inline Build/Projects/FrameworkReferences.definition]: sdl mixer, audio

**Comment by cra0zy** [comment]
> What platforms has this been tested on?

I tested it on Linux and WindowsGL, and I know it works on Mac because of #3119.

> Also after we merge this, can you follow up with a second PR organizing:
> 
> Axis
> Capabilities
> Input
> PadConfig
> Joystick
> Stick
> These types seem to be internals to the OpenTK GamePad implementation. I don't think we want them public. We also probably want to move them into some other namespace like Microsoft.Xna.Framework.Input.OpenTK or something.

Gamepad uses OpenTK.Gamepad class, it doesn't use any of those classes. I thought it would be nice to leave those Joystick classes around, gives options for some lower level joystick implementation.

**Comment by cra0zy** [comment]
Now that I think about it, maybe I should remove Joystick classes completely, GamePad.OpenTK doesn't use them, and Xna never had them.

File: Build/Projects/FrameworkReferences.definition (hunk 1 above)
Comment (tomspilman) [inline Build/Projects/FrameworkReferences.definition]: Ah... so we are waiting on https://github.com/mono/MonoGame/issues/3625.

**Comment by tomspilman** [comment]
> Gamepad uses OpenTK.Gamepad class, it doesn't use any of those classes.

Well how about we unhook them from "MonoGame.Framework.definition" in this PR then?  This way they won't show up in the docs and won't confuse people for now.

Separately we can start some discussion on if we want to support a new API beyond GamePad for more complex input devices like many axis joysticks.

**Comment by cra0zy** [comment]
Sounds good to me.

**Comment by tomspilman** [comment]
Merging... thanks!

**Pull request merged**